runtime/src/platform_utils.cpp

```diff
@@ -0,0 +1,49 @@
+#include "platform_utils.hpp"
+
+#if __linux__
+#include <sched.h>
+
+int my_sched_getcpu() {
+    return ::sched_getcpu();
+}
+
+#elif __APPLE__
+#include <sys/sysctl.h>
+#include <pthread.h>
+
+int my_sched_getcpu() {
```

Comment:
Maybe... `platform_sched_getcpu()`?

Reply:
Linux's is called `sched_getcpu()` already, right? What if we just define it for APPLE? Then it would be magically available on both, wouldn't it?

As long as we implement the same interface

Reply:
I tried that way but was facing warning's from gcc on infinite recursion, even though it was two different scoped functions, you can see in previous version, it was 'sched_getcpu()' with consistent interface, but i replaced it altogether to remove those warnings. @Officeyutong  @yunwei37 what do you suggest?

Reply:
using `get_currentcpu()` in all files now.